morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -463,13 +399,30 @@ public void ensureIndexes(final boolean background) {
         return getByKeys(null, keys);
     }
 
+    private DBCollection getCollection(final Object obj) {
+        if (obj == null) {
+            return null;
```

Comment:
It's (mostly) not ok for user input to be null.  but this is the first time it gets checked on any of the paths to this point.